FindElementInInfiniteArray.java

```diff
@@ -0,0 +1,64 @@
+package s30Coding;
+
+public class FindElementInInfiniteArray {
+
+	public static void main(String[] args) {
+		FindElementInInfiniteArray f = new FindElementInInfiniteArray();
+		int[] nums = new int[] {10,20,30,45,50,100};
+		System.out.println(f.search(nums,99));
+	}
+		// TODO Auto-generated method stub
+		public int search(int[] nums, int target) {
+			
+			
+			int length = 1;
+			
+			//calculate the high end for the array search since we don't know the length
+			try { while(target > nums[length])
+		     {
+		    	 length = length+1;
+		     }
+			}
+			catch(ArrayIndexOutOfBoundsException e){
+				throw e;
+			}
+			
+		    
+	        int result = search(nums, 0, length, target);
+	        return result;
+	       
+	    }
+	   
+	      static int search(int num[], int low, int high, int target)
+	    {
+	         
+	        if (low > high) 
+	            return -1;
+	       
+	        int mid = (low+high)/2;
```

Comment:
In order to not face Integer overflow error, we generally calculate mid as

 mid = low + (high-low)/2

Reply:
Yeah, I got. I learned about it yesterday during the lecture. I will update it.